omf-python/src/file/reader.rs

```diff
@@ -0,0 +1,277 @@
+use crate::array::{
+    PyBooleanArray, PyBoundaryArray, PyColorArray, PyGradientArray, PyImageArray, PyIndexArray,
+    PyNameArray, PyNumberArray, PyScalarArray, PySegmentArray, PyTexcoordArray, PyTextArray,
+    PyTriangleArray, PyVectorArray, PyVertexArray,
+};
+use crate::errors::OmfException;
+use crate::validate::PyProblem;
+use crate::PyProject;
+use omf::error::Error::IoError;
+use omf::file::{Limits, Reader};
+use omf::Color;
+use pyo3::prelude::*;
+use pyo3::types::PyBytes;
+use pyo3_stub_gen::derive::*;
+use std::fs::File;
+
+#[gen_stub_pyclass]
+#[pyclass(name = "Limits")]
+/// Memory limits for reading OMF files.
+pub struct PyLimits {
+    #[pyo3(get, set)]
+    pub json_bytes: Option<u64>,
+    #[pyo3(get, set)]
+    pub image_bytes: Option<u64>,
+    #[pyo3(get, set)]
+    pub image_dim: Option<u32>,
+    #[pyo3(get, set)]
+    pub validation: Option<u32>,
+}
+
+#[gen_stub_pymethods]
+#[pymethods]
+impl PyLimits {
+    #[new]
+    pub fn new() -> PyResult<Self> {
+        let limits = Limits::default();
+        Ok(PyLimits {
+            json_bytes: limits.json_bytes,
+            image_bytes: limits.image_bytes,
+            image_dim: limits.image_dim,
+            validation: limits.validation,
+        })
+    }
+}
+
+#[gen_stub_pyclass]
+#[pyclass(name = "Reader")]
+/// OMF reader object.
+///
+/// Typical usage pattern is:
+///
+/// - Create the reader object.
+/// - Optional: retrieve the file version with `reader.version()`.
+/// - Optional: adjust the limits with `reader.set_limits(...)`.
+/// - Read the project from the file with `reader.project()`.
+/// - Iterate through the project's contents to find the elements and attributes you want to load.
+/// - For each of those items load the array or image data.
+///
+/// **Warning:**
+///     When loading arrays and images from OMF files, beware of "zip bombs"
+///     where data is maliciously crafted to expand to an excessive size when decompressed,
+///     leading to a potential denial of service attack.
+///     Use the limits provided check arrays sizes before allocating memory.
```

Comment:
Might be an error in where this was copied from.
```suggestion
///     Use the limits provided and check arrays sizes before allocating memory.
```